contrib_bots/lib/myanimelist.py

```diff
@@ -0,0 +1,83 @@
+from __future__ import print_function
+import requests
+from xml.etree import ElementTree
+
+# Signup for a MAL account here https://myanimelist.net/register.php?from=%2F
+USERNAME = '[PLEASE CHANGE]'
+PASSWORD = '[PLEASE CHANGE]'
+
+# See docs.md for instructions on running this code.
+MAL_SEARCH_URL = "https://myanimelist.net/api/anime/search.xml?q="
+MAL_PAGE_URL = "https://myanimelist.net/anime/"
+
+class LinksHandler(object):
+    def search(self, query):
+        response = requests.get(MAL_SEARCH_URL + query, auth=(USERNAME, PASSWORD))
+        try:
+            xml = ElementTree.fromstring(response.content).find('entry')
+            contentDict = {
+                'id': '',
+                'score': '',
+                'type': '',
+                'episodes': '',
+                'status': '',
+                'title': '',
+            }
+            for key, value in contentDict.items():
+                contentDict[key] = xml.find(key).text
+                if not contentDict[key]:
+                    return None, False
+            return contentDict, True
+        except:
+            return None, False
+    '''
+    This plugin responds with the link of to the anime's
+    myanimelist page and some details along with it
+    It looks for messages starting with '@mal'
+    '''
+
+    def usage(self):
+        if USERNAME == "[PLEASE CHANGE]" or PASSWORD == "[PLEASE CHANGE]":
```

Comment:
Good idea. However, looking at `contrib_bots/run.py` it seems that `usage` runs only when `quiet` option is not set so this check will be bypassed when `--quiet` is used. You can move this check in the global scope that way whenever the module is imported your code will run.